src/main/java/org/sagebionetworks/bridge/models/schedules2/timelines/Scheduler.java

```diff
@@ -0,0 +1,159 @@
+            ScheduledSession.Builder scheduledSession = new ScheduledSession.Builder();
+            scheduledSession.withRefGuid(session.getGuid());
+            scheduledSession.withStartDay(startDay);
+            scheduledSession.withEndDay(endDay);
+            scheduledSession.withStartTime(window.getStartTime());
+            scheduledSession.withExpiration(window.getExpiration());
+            scheduledSession.withPersistent(window.isPersistent());
+            // If we have a delay under one day, and it's the first day, then skip the startTime 
+            // and delay for the indicated time before showing.
+            if (startDay == 0 && delay != null && delay.toStandardDays().getDays() == 0) {
+                scheduledSession.withDelayTime(delay);
+                scheduledSession.withStartTime(null);
+            }
+            
+            String sessionInstanceGuid = generateSessionInstanceGuid(
+                    schedule.getGuid(), session.getGuid(), window.getGuid(), occurrenceCount);
+            scheduledSession.withInstanceGuid(sessionInstanceGuid);
+
+            // The position of an assessment in a session is used to differentiate repeated assessments
+            // in a single session. Assessments can be configured differently, but if they are exactly
+            // the same, we still generate a unique ID.
+            ids.clear();
+            for (AssessmentReference ref : session.getAssessments()) {
+                ids.add(ref.getGuid());
+                
+                String asmtInstanceGuid = generateAssessmentInstanceGuid(schedule.getGuid(), session.getGuid(),
+                        window.getGuid(), occurrenceCount, ref.getGuid(), ids.count(ref.getGuid()));
+             
+                AssessmentInfo asmtInfo = AssessmentInfo.create(ref);
+                builder.withAssessmentInfo(asmtInfo);
+                
+                ScheduledAssessment schAsmt = new ScheduledAssessment(asmtInfo.getKey(), asmtInstanceGuid, ref);
+                scheduledSession.withScheduledAssessment(schAsmt);
+            }
+            builder.withScheduledSession(scheduledSession.build());
+      
+            startDay += intervalInDays;
+            occurrenceCount++;
+            
+        } while(intervalInDays > 0 && occurrenceCount < occurrenceMax && startDay < studyLengthInDays);
+    }
+    
+    /**
+     * Create a unique GUID for a session instance that is the same each time you calculate a Timeline
+     * from a Schedule. It should look like a GUID and not a compound identifier, as client developers 
+     * have (in the past) parsed compound identifiers, and we want to discourage this.
+     */
+    String generateSessionInstanceGuid(String scheduleGuid, String sessionGuid, String windowGuid,
+            int windowOccurrence) {
+        return windowGuid.substring(0, 8) + Integer.toHexString(windowOccurrence) 
+            + sessionGuid.substring(0, 8) + scheduleGuid.substring(0, 6);
+    }
```

Comment:
Maybe make these a little longer to ensure you have the same amount of randomness.

Reply:
Converted to using 128 byte murmur3 algorithm, with Base 64 encoding to reduce the length of the result, across the full value of the GUIDs.